pkg/reconciler/common/reconcile.go

```diff
@@ -45,11 +48,28 @@ var knativeServingAnnotations = []string{
 	serving.UpdaterAnnotation,
 }
 
-// AdapterDeploymentBuilderFunc builds a Deployment object for a source's adapter.
-type AdapterDeploymentBuilderFunc func(sinkURI *apis.URL) *appsv1.Deployment
+// RBACOwnersLister returns a list of OwnerRefable to be set as a the
+// OwnerReferences metadata attribute of a ServiceAccount.
+type RBACOwnersLister interface {
+	RBACOwners(namespace string) ([]kmeta.OwnerRefable, error)
+}
+
+// AdapterDeploymentBuilder provides all the necessary information for building
+// objects related to a source's adapter backed by a Deployment.
+type AdapterDeploymentBuilder interface {
+	RBACOwnersLister
+	BuildAdapter(src v1alpha1.EventSource, sinkURI *apis.URL) *appsv1.Deployment
+}
+
+// AdapterServiceBuilder provides all the necessary information for building
+// objects related to a source's adapter backed by a Knative Service.
+type AdapterServiceBuilder interface {
+	RBACOwnersLister
+	BuildAdapter(src v1alpha1.EventSource, sinkURI *apis.URL) *servingv1.Service
+}
```

Comment:
Here are the interfaces each `Reconciler` must implement (one or the other, not both).

* `RBACOwners` returns the list of objects that should be set as owner on the kind's ServiceAccount. All reconcilers return the entire contents of their Lister in the current implementation (all instances of the source type), so that the Kubernetes garbage collector kicks in once that list becomes empty (all instances of the sources are deleted).
* `BuildAdapter` returns an adapter object for the current source and sink (same as before, just inside an interface).